### PR TITLE
[*] UploadServices & init useNewUploadContext experiment

### PR DESCRIPTION
**Pull request by talevy17**

### CODE REVIEW

File: examples/storybook/src/shared/utils/fileUploadUtil.ts [hunk 1]
```diff
@@ -9,6 +9,8 @@ import { testImages, testWixVideos } from './mock';
 import { mockAudioData } from './mockAudioData';
 import { FILE_URLS } from './fileMockURL';
 
+const error = { key: 0, msg: 'error uploading file' };
```
Comment (danarish4) [inline examples/storybook/src/shared/utils/fileUploadUtil.ts:12]: is it used?

Comment (talevy17) [reply examples/storybook/src/shared/utils/fileUploadUtil.ts:12]: only when I'm debugging error handling, can be removed (:

**Review by danarish4 — CHANGES_REQUESTED** [review]

File: packages/plugin-commons/web/src/Utils/generateInsertPluginButtonProps.ts [hunk 2]
```diff
@@ -178,28 +185,60 @@ export function generateInsertPluginButtonProps({
...
       commonPubsub.set('initialState_' + blockKey, state);
-    });
+    };
+    return handleFileChange(files, updateEntity);
+  }
+
+  function uploadFiles(files: File[]) {
+    const getUploader = button.getUploader;
+    const mediaPluginService = button.mediaPluginService;
+    if (getUploader && mediaPluginService) {
```
[the remark this answers is not in the document]
Comment (talevy17) [reply packages/plugin-commons/web/src/Utils/generateInsertPluginButtonProps.ts:204]: we need it for type checking (ts can get annoying)

File: packages/plugin-gallery/web/src/components/gallery-settings-modal.js [hunk 3]
```diff
@@ -43,22 +50,56 @@ class ManageMediaSection extends Component {
...
+      : this.handleFilesAdded;
+    helpers.handleFileSelection(index, multiple, handleFilesAdded, deleteBlock, data);
   };
 
   handleFilesAdded = (...args) => {
     const { store } = this.props;
     store.getBlockHandler('handleFilesAdded')?.(...args);
   };
 
+  handleFilesAddedWithIndex = index => {
```
Comment (talevy17) [inline packages/plugin-gallery/web/src/components/gallery-settings-modal.js:86]: use UpdateService instead.
updateData is deprecated.

File: packages/plugin-gallery/web/src/components/gallery-settings-modal.js [hunk 4]
```diff
@@ -318,73 +362,87 @@ export class GallerySettingsModal extends Component {
...
-      <Tab
-        label={this.tabName('advanced_settings', this.props.t)}
-        value={'advanced_settings'}
-        theme={this.props.theme}
-      >
-        <AdvancedSettingsSection
+        >
+          {this.props.experiments.useNewUploadContext?.enabled ? (
+            <UploadServiceContext.Consumer>
+              {({ uploadService }) => (
```
Comment (talevy17) [inline packages/plugin-gallery/web/src/components/gallery-settings-modal.js:392]:
```suggestion
              {({ uploadService, updateService }) => (
```
```suggestion
              {({ uploadService }) => (
```

Comment (talevy17) [inline packages/plugin-gallery/web/src/components/gallery-settings-modal.js:392]:
```suggestion
              {({ uploadService, updateService }) => (
```
```suggestion
              {({ uploadService }) => (
```

File: packages/plugin-gallery/web/src/gallery-component.jsx [hunk 5]
```diff
@@ -28,7 +28,9 @@ function GalleryComponent(props) {
         blockKey={props.block.getKey()}
         itemOverlayElement={renderMobileNativeLoader}
       />
-      {!error && isLoading && <Loader type={'medium'} percent={loadingPercentage} />}
+      {!error && (isLoading || componentData.loading) && (
```
Comment (talevy17) [inline packages/plugin-gallery/web/src/gallery-component.jsx:31]: check that items cannot be saved with 'loading' in the content

Comment (talevy17) [inline packages/plugin-gallery/web/src/gallery-component.jsx:31]: check that items cannot be saved with 'loading' in the content

File: packages/plugin-gallery/web/src/toolbar/galleryPluginService.ts [hunk 6]
```diff
@@ -0,0 +1,112 @@
...
+};
+
+export class GalleryPluginService implements IMediaPluginService {
+  createLoadingData(
+    file: File,
+    _: unknown,
+    componentData: ComponentData,
+    fileState?: Record<string, string | number>
+  ) {
+    // const fileType = getGalleryFileType(file.type);
```
Comment (talevy17) [inline packages/plugin-gallery/web/src/toolbar/galleryPluginService.ts:70]: TODO: revert when PG fix bug.

Comment (talevy17) [inline packages/plugin-gallery/web/src/toolbar/galleryPluginService.ts:70]: TODO: revert when PG fix bug.

File: packages/plugin-image/web/src/image-component.jsx [hunk 7]
```diff
@@ -63,7 +63,7 @@ class ImageComponent extends React.Component {
           componentData={componentData}
           className={className}
           isLoading={this.props.isLoading}
-          dataUrl={this.props.tempData?.dataUrl}
+          dataUrl={this.props.tempData?.dataUrl || componentData.dataUrl}
```
Comment (talevy17) [inline packages/plugin-image/web/src/image-component.jsx:66]:
```suggestion
          dataUrl={this.props.tempData?.dataUrl || componentData.tempData.dataUrl}
```
```suggestion
          dataUrl={this.props.tempData?.dataUrl || componentData.dataUrl}
```

Comment (talevy17) [inline packages/plugin-image/web/src/image-component.jsx:66]:
```suggestion
          dataUrl={this.props.tempData?.dataUrl || componentData.tempData.dataUrl}
```
```suggestion
          dataUrl={this.props.tempData?.dataUrl || componentData.dataUrl}
```

File: packages/plugin-video/web/src/toolbar/videoUploadModal.jsx [hunk 8]
```diff
@@ -21,6 +22,25 @@ export default class VideoUploadModal extends Component {
         onReplace(obj, this.blockKey);
       }
     };
+    this.VideoPluginService = new VideoPluginService();
+    this.Uploader = new Uploader(props.handleFileUpload);
+    this.uploadContextOnConfirm = (data, isCustomVideo) => {
+      if (onConfirm) {
+        const componentData = this.VideoPluginService.createPluginData(
+          { data },
+          this.props.componentData
```
Comment (talevy17) [inline packages/plugin-video/web/src/toolbar/videoUploadModal.jsx:31]: can't use componentData directly.

get from Pubsub or from getComponentData (withModalBaseActions).

Comment (talevy17) [inline packages/plugin-video/web/src/toolbar/videoUploadModal.jsx:31]: can't use componentData directly.

get from Pubsub or from getComponentData (withModalBaseActions).

File: packages/ricos-common/web/src/UploadContext/ErrorNotifier.tsx [hunk 9]
```diff
@@ -0,0 +1,73 @@
+import React, { Component, Suspense } from 'react';
+import type { IMessage, INotifier } from 'ricos-types';
+import { RicosTranslate } from 'wix-rich-content-common';
+import { getLangDir } from '../utils';
+const ErrorToast = React.lazy(() => import('wix-rich-content-editor/libs/ErrorToast'));
```
Comment (justame) [inline packages/ricos-common/web/src/UploadContext/ErrorNotifier.tsx:5]: the name ErrorToast indicates that it's only for errors. if so, great.

**Pull request merged**